### PR TITLE
Return introspection document in stable order

### PR DESCRIPTION
**Pull request by nyergler**
This avoids spurious changes when generating client code using something like graphql-codegen.

Describe your PR and link to any relevant issues. 

I have:
 - [ ] Added tests covering the bug / feature (see [testing](https://github.com/99designs/gqlgen/blob/master/TESTING.md))
 - [X] Updated any relevant documentation (see [docs](https://github.com/99designs/gqlgen/tree/master/docs/content))

### CODE REVIEW

**Comment by xzyfer** [comment]
🙏

**Comment by nyergler** [comment]
I wonder if it's possible to get this merged? It resolves #1426

**Comment by xzyfer** [comment]
Fwiw we've been running this in production for the last week

File: graphql/introspection/schema.go [hunk 1]
```diff
@@ -11,12 +12,20 @@ type Schema struct {
 }
 
 func (s *Schema) Types() []Type {
-	types := make([]Type, 0, len(s.schema.Types))
+	typIndex := map[string]Type{}
```
Comment (wilhelmeek) [inline graphql/introspection/schema.go:15]: Is the spelling here intentional? 
```suggestion
	typeIndex := map[string]Type{}
```

**Review by wilhelmeek — APPROVED** [review]

File: graphql/introspection/schema.go (hunk 1 above)
Comment (nyergler) [reply graphql/introspection/schema.go]: Nope, fixed!

**Comment by mtibben** [comment]
Thanks @nyergler and apologies for the delay

**Pull request merged**